### ISSUE TITLE
Incorrect buffer position passed to a provider when bracket matcher added a closed parethesis

### ISSUE THREAD

**Issue by abe33** [atom/autocomplete-plus#954]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

The buffer position passed to the provider doesn't match the cursor position when the request occurs concurrently with `bracket-matcher` completions. 

### Steps to Reproduce

1. In the devtools console enters the following sample code: ```atom.packages.serviceHub.provide('autocomplete.provider', '4.0.0', {
    selector: '*',
    getSuggestions(params) {console.log(params.bufferPosition) }
});```
2. In a new file, type `foo(`
3. Notice that the buffer position in the provider function is now `0, 5` while the cursor is actually at `0, 4`.

**Expected behavior:** 

The position in the provider parameters should be the position inside the parentheses.

**Actual behavior:**

We receive the position after the second parenthesis.

**Reproduces how often:** 100%

### Versions

```
Atom    : 1.26.0-dev-0b59e5f
Electron: 1.7.11
Chrome  : 58.0.3029.110
Node    : 7.9.0
```

```
apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.26.0-dev-0b59e5f
python 2.7.13
git 2.2.1
```

**Comment by iolsen** [comment]
Putting this on prioritized for investigation. If this is a deep, dark hole we may not fix it promptly.

**Comment by smashwilson** [comment]
I can't seem to reproduce this:

<img width="467" alt="screen shot 2018-03-19 at 4 18 16 pm" src="https://user-images.githubusercontent.com/17565/37619904-288055ec-2b91-11e8-854c-1b5918603ae4.png">

I don't see any recent work here that would fix this, though. Can you still reproduce on master? Do you have any settings in autocomplete-plus or bracket-matcher that might effect this?

**Comment by smashwilson** [comment]
... Wait. My package might not have been up-to-date. I'll check again tomorrow.

**Comment by smashwilson** [comment]
Okay, yup, can repro 👍

**Comment by smashwilson** [comment]
Fixed by atom/bracket-matcher#347 🔧
